File: source/frontend/src/features/management/activities/models/ManagementActivityFilterModel.ts
import { MultiSelectOption } from '@/interfaces/MultiSelectOption';
import { Api_ManagementActivityFilter } from '@/models/api/ManagementActivityFilter';

export class ManagementActivityFilterModel {
  searchBy = 'address';
  pin = '';
  pid = '';
  regionCodes = [];
  address = '';
  fileNameOrNumberOrReference = '';
  activityStatusCode = '';
  activityTypeCode = '';
  projectNameOrNumber = '';
  managementFileStatusCode = '';
  managementFilePurposeCode = '';

  constructor(initialRegions: MultiSelectOption[] = []) {
    this.regionCodes = initialRegions;
  }

  toApi(): Api_ManagementActivityFilter {
    return {
      searchBy: this.searchBy,
      pin: this.pin,
      pid: this.pid,
      regionCodes: this.regionCodes?.map(x => x.id) ?? [],
      address: this.address,
      fileNameOrNumberOrReference: this.fileNameOrNumberOrReference,
      activityStatusCode: this.activityStatusCode,
      activityTypeCode: this.activityTypeCode,
      projectNameOrNumber: this.projectNameOrNumber,
      managementFileStatusCode: this.managementFileStatusCode,
      managementFilePurposeCode: this.managementFilePurposeCode,
    };
  }

  static fromApi(
    base: Api_ManagementActivityFilter,
    userRegions: MultiSelectOption[],
  ): ManagementActivityFilterModel {
    const newModel = new ManagementActivityFilterModel();
    newModel.searchBy = base.searchBy ?? 'address';
    newModel.pin = base.pin ?? '';
    newModel.pid = base.pid ?? '';
    newModel.regionCodes = userRegions ?? [];
    newModel.address = base.address ?? '';
    newModel.fileNameOrNumberOrReference = base.fileNameOrNumberOrReference ?? '';
    newModel.activityStatusCode = base.activityStatusCode ?? '';
    newModel.activityTypeCode = base.activityTypeCode ?? '';
    newModel.projectNameOrNumber = base.projectNameOrNumber ?? '';
    newModel.managementFileStatusCode = base.managementFileStatusCode ?? '';
    newModel.managementFilePurposeCode = base.managementFilePurposeCode ?? '';

    return newModel;
  }
}

File: source/backend/api/Areas/Management/Models/ManagementActivityFilterModel.cs
using System;
using System.Collections.Generic;
using Pims.Core.Extensions;
using Pims.Dal.Entities.Models;

namespace Pims.Api.Areas.Management.Models
{
    public class ManagementActivityFilterModel : PageFilter
    {
        /// <summary>
        /// get/set - The pid identifier to search by.
        /// </summary>
        public string Pid { get; set; }

        /// <summary>
        /// get/set - The pin identifier to search by.
        /// </summary>
        public string Pin { get; set; }

        /// <summary>
        /// get/set - The address to search by.
        /// </summary>
        public string Address { get; set; }

        /// <summary>
        /// get/set - The management file name or the file number or the legacy reference number, search for all simultaneously.
        /// </summary>
        public string FileNameOrNumberOrReference { get; set; }

        /// <summary>
        /// get/set - The Activity Type.
        /// </summary>
        public string ActivityTypeCode { get; set; }

        /// <summary>
        /// get/set - The Activity Status Code.
        /// </summary>
        public string ActivityStatusCode { get; set; }

        /// <summary>
        /// get/set - The MOTI project name or the project number, search for both simultaneously.
        /// </summary>
        public string ProjectNameOrNumber { get; set; }

        public ManagementActivityFilterModel()
        {
        }

        public ManagementActivityFilterModel(Dictionary<string, Microsoft.Extensions.Primitives.StringValues> query)
            : base(query)
        {
            var filter = new Dictionary<string, Microsoft.Extensions.Primitives.StringValues>(query, StringComparer.OrdinalIgnoreCase);

            Pid = filter.GetStringValue(nameof(Pid));
            Pin = filter.GetStringValue(nameof(Pin));
            Address = filter.GetStringValue(nameof(Address));
            FileNameOrNumberOrReference = filter.GetStringValue(nameof(FileNameOrNumberOrReference));
            ActivityTypeCode = filter.GetStringValue(nameof(ActivityTypeCode));
            ActivityStatusCode = filter.GetStringValue(nameof(ActivityStatusCode));
            ProjectNameOrNumber = filter.GetStringValue(nameof(ProjectNameOrNumber));
        }

        /// <summary>
        /// Convert to a ManagementActivityFilter.
        /// </summary>
        /// <param name="model"></param>
        public static explicit operator ManagementActivityFilter(ManagementActivityFilterModel model)
        {
            var filter = new ManagementActivityFilter
            {
                Page = model.Page,
                Quantity = model.Quantity,

                Pid = model.Pid,
                Pin = model.Pin,
                Address = model.Address,
                FileNameOrNumberOrReference = model.FileNameOrNumberOrReference,
                ActivityTypeCode = model.ActivityTypeCode,
                ActivityStatusCode = model.ActivityStatusCode,
                ProjectNameOrNumber = model.ProjectNameOrNumber,

                Sort = model.Sort,
            };

            return filter;
        }
    }
}
